### PR TITLE
Use an extra socket in dmtcp_get_coord_ckpt_dir().

### PR DESCRIPTION
**Pull request by jiajuncao**
Fix issue #438.

### CODE REVIEW

File: src/coordinatorapi.cpp [hunk 1]
```diff
@@ -386,16 +386,26 @@ string CoordinatorAPI::getCoordCkptDir(void)
...
+  sock >> msg;
   msg.assertValid();
-  JASSERT(msg.type == DMT_GET_CKPT_DIR_RESULT) (msg.type);
+  JASSERT(msg.type == DMT_GET_COORD_CKPT_DIR_RESULT) (msg.type);
+
+  if (msg.extraBytes > 0) {
+    sock.readAll(buf, msg.extraBytes);
+  }
+  else {
+    JWARNING("Failed to get the coordinator checkpoint dir. "
```
Comment (rohgarg) [inline src/coordinatorapi.cpp]: `JWARNING()` has the same syntax as `JASSERT()`. Please change this to:
`JWARNING(false).Text("...")`

File: src/dmtcpplugin.cpp [hunk 2]
```diff
@@ -199,7 +199,9 @@ EXTERNC int dmtcp_set_ckpt_dir(const char* dir)
 EXTERNC const char* dmtcp_get_coord_ckpt_dir(void)
 {
   static string dir;
+  dmtcp_disable_ckpt();
```
Comment (rohgarg) [inline src/dmtcpplugin.cpp]: I believe we should be using the helper macros instead of calling the functions directly.

Comment (jiajuncao) [inline src/dmtcpplugin.cpp]: You mean DMTCP_PLUGIN_DISABLE_CKPT()? I assume this is primarily used in plugins.

File: src/dmtcpplugin.cpp [hunk 3]
```diff
@@ -199,7 +199,10 @@ EXTERNC int dmtcp_set_ckpt_dir(const char* dir)
 EXTERNC const char* dmtcp_get_coord_ckpt_dir(void)
```
Comment (rohgarg) [inline src/dmtcpplugin.cpp]: Could you please document that the API returns a zero-length string, "", when a checkpoint is in progress?

File: src/dmtcpplugin.cpp [hunk 4]
```diff
@@ -199,7 +199,10 @@ EXTERNC int dmtcp_set_ckpt_dir(const char* dir)
...
 {
   static string dir;
```
Comment (rohgarg) [inline src/dmtcpplugin.cpp]: Is there any reason we want a static string object here? If not, can we get rid of this static string object?

**Pull request merged**